### PR TITLE
Add support to Kubernetes 1.16, 1.17 and 1.18

### PR DESCRIPTION
**Pull request by angelbarrera92**
Hi team

This PR adds support to 1.16, 1.17 and 1.18 to this module, also i included the new linter to the pipeline. This caused to modify some files in the repository. (sorry for that).

Also, i updated the terraform azure provider of the azure terraform velero module and i have changed the resource group used during the e2e-testing to avoid race conditions while creating role bindings.

Thanks!

### CODE REVIEW

File: .rules/.tflint.hcl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+config {
+  ignore_module = {
+    "../../modules/aws-velero" = true
+    "../../modules/gcp-velero" = true
+    "../../modules/azure-velero" = true
+  }
+}
```
Comment (iknite) [inline .rules/.tflint.hcl:7]: Carriage return :)

**Comment by angelbarrera92** [comment]
[1.16, 1.17 and 1.18 support for core module: dr](https://trello.com/c/9wrfiNfy/266-116-117-and-118-support-for-core-module-dr)

File: docs/releases/v1.2.0.md [hunk 2]
```diff
@@ -73,12 +73,12 @@ $ kubectl delete clusterrolebindings -l component=ark
...
 $ kubectl delete ServiceMonitor ark -n kube-system
 
-# Remove old kustomization.yaml.old file 
+# Remove old kustomization.yaml.old file
 $ rm kustomization.yaml.old
 ```
 
 
-#### Links
+#### Links - Ark to velero
```
Comment (ralgozino) [inline docs/releases/v1.2.0.md:81]: velero -> Velero

File: docs/releases/v1.2.0.md [hunk 3]
```diff
@@ -108,10 +108,11 @@ Then apply the configuration:
 
 ```bash
 $ kustomize build | kubectl apply -f -
+# omitted output
 ```
 
 
-#### Links
+#### Links Verlero 1.0 to 1.2
```
Comment (ralgozino) [inline docs/releases/v1.2.0.md:115]: Verlero -> Velero

File: docs/releases/v1.4.0.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Releases notes
+
+## Changelog
+
+Changes between `1.3.1` and this release: `1.4.0`.
+
+- Support 1.16, 1.17 and 1.18 Kubernetes Version
```
Comment (ralgozino) [inline docs/releases/v1.4.0.md:7]: Support 1.16 [...] -> Support for 1.16[...]

**Pull request merged**